### DISCUSSION TITLE
When do you acknowledge contributors?

### DISCUSSION THREAD

**Discussion by elecbug** [libp2p/go-libp2p#3009] [Q&A] [answered]
I have finded the problem and suggest a way to fix that. and submited a PR for that, but unlike the general merge system, I am a little confused by the fact that the problem is solved outside of my PR with a only simple (i.e., not including me) comment in other PR. 

Am I missing something? I think this kind of non-recognition of even small contributions can be demoralizing to contributors.

[It's my PR](https://github.com/libp2p/go-libp2p/pull/2996)

Is there any way we can ensure small contributions are better recognized in the future?

**Comment by MarcoPolo** [comment, marked as answer]
Apologies @elecbug. I'll call out your contribution in the release notes for the next version in which we'll release this fix.

In general though, I wouldn't call out a bug report by itself unless it's substantial. If you're looking to contribute, may I humbly suggest helping review PRs? I'd acknowledge the contribution as a Git commit footer message at least (something like reviewed-by: foo).

In the case of pushing a new PR, at minimum it should include tests. It should include a motivation for the change. And it should be mindful of future maintenance efforts. It's better to start discussion on an issue, and then create a PR.

Again, no slight intended here. Just trying to make a great and useful library. Thanks for understanding.

**Reply by elecbug** [reply]
No. Rather, I apologize. I didn't feel any insult or anger in your response. Rather, I appreciate it.

I recently stepped into the open-source community as a developer, and I was trying to contribute to several projects with some simple modifications.

In a way, I think I was feeling satisfied with myself by being recognized as a contributor. I think it's natural to acknowledge useful and reasonable PR.

I think I went into the open-source community and did something wrong because I didn't understand the proper rules or manners and I was trying to hastily achieve something.

I will remember your answer and philosophy for a long time from now on.

Thank you for always working hard for our library.
